ledger/internal/cow_creatables.go

```diff
@@ -0,0 +1,216 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package internal
+
+import (
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/basics"
+)
+
+func (cs *roundCowState) MinBalance(addr basics.Address, proto *config.ConsensusParams) (res basics.MicroAlgos, err error) {
+	acct, err := cs.lookup(addr) // pending rewards unneeded
+	if err != nil {
+		return
+	}
+	return acct.MinBalance(proto), nil
+}
+
+func (cs *roundCowState) TotalAppParams(creator basics.Address) (int, error) {
+	acct, err := cs.lookup(creator)
+	if err != nil {
+		return 0, err
+	}
+	return len(acct.AppParams), nil
+}
+func (cs *roundCowState) TotalAppLocalState(addr basics.Address) (int, error) {
+	acct, err := cs.lookup(addr)
+	if err != nil {
+		return 0, err
+	}
+	return len(acct.AppLocalStates), nil
+}
+func (cs *roundCowState) TotalAssetHolding(addr basics.Address) (int, error) {
```

Comment:
Since I previously said I'd prefer AssetHoldingCount, I suppose I should say I agree they should all be uniform.  AppParamsCount, AppLocalStateCount seem ok.

Reply:
This `Total*` name comes from existing `TotalAppSchema/TotalExtraAppPages` and upcoming `TotalAssets/TotalAppParams/TotalAppLocalStates` fields

Reply:
But notice that TotalAppSchema and TotalExtraPages are actually _totaling_.  They add up the values inside the schema, or the values of the epp fields.  These don't total up anything. But, since one is inclined to think that a Total*() method is totaling things up, when you see TotalAssetHolding it seems like it must be totaling up some assets.